django/contrib/gis/db/models/functions.py

```diff
@@ -421,6 +422,37 @@ class Intersection(OracleToleranceMixin, GeomOutputGeoFunc):
     geom_param_pos = (0, 1)
 
 
+@BaseSpatialField.register_lookup
+class GeometryType(GeoFuncMixin, Transform):
+    function = "GeometryType"
+    output_field = CharField()
+    lookup_name = "geom_type"
+
+    def as_mysql(self, compiler, connection, **extra_context):
+        lhs, params = compiler.compile(self.lhs)
+        sql = f"ST_GeometryType({lhs})"
+        return sql, params
```

Comment:
This is unnecessary, MySQL/MariaDB GIS functions are automatically prefixed with `ST_`.
```suggestion
```